qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -363,7 +363,86 @@ def validate_parameter(self, parameter):
 
 
 def initialize(self, params, qubits=None):
-    """Apply initialize to circuit."""
+    """Qubit initializalition is done by appending instructions to the quantum circuit (by
+    calling Initialize(params)) and the qubits we wish to initialize. Note that the
+    qubits are first set to `|0>` and then the desired state is achieved by applying
+    a state preparing unitary.
+
+    Args:
+        params (str or list or int):
+            * str: labels of basis states of the Pauli eigenstates Z, X, Y. See
+                :meth:`~qiskit.quantum_info.states.statevector.Statevector.from_label`.
+                Notice the order of the labels is reversed with respect to the qubit index to
+                be applied to. Example label '01' initializes the qubit zero to `|1>` and the
+                qubit one to `|0>`.
+            * list: vector of complex amplitudes to initialize to.
+            * int: an integer that is used as a bitmap indicating which qubits to initialize
+               to `|1>`. Example: setting params to 5 would initialize qubit 0 and qubit 2
+               to `|1>` and qubit 1 to `|0>`.
+        qubits (QuantumRegister or int):
+            * QuantumRegister: A list of qubits to be initialized [Default: None].
+            * int: Index of qubit to initialized [Default: None].
+
+    Returns:
+        qiskit.circuit.Instruction: a handle to the instruction that was just initialized
+
+    Examples:
+        Prepare a qubit in the anti-symmetric state 1/sqrt(2)(`|0>` - `|1>`).
```

Comment:
If we use a raw docstring (start the docstring with `r"""` instead of just `"""`) we can use tex math 🙂 Also I'd be cautious with the term "anti-symmetric" since you'd have to specify with respect to what axis/operation it is symmetric. Since it's not crucial information we can just remove it.
```suggestion
        Prepare a qubit in the state :math:`(|0\rangle - |1\rangle) / \sqrt{2}`.
```